CenterForOpenScience/osf.io
# [Feature] Add button to reset client application secret

Purpose
======
Allow users with developer applications to reset their client secrets in the event of a security compromise (e.g. committing it to github, unauthorized access discovered, etc.).

Changes
=======
Edits to:
- serializers: `*ApplicationBaseSerializer` created to generalize some repeated code
- models: function added to reset secret
- urls: endpoint added to reset secret
- views: mixin added for generalization, view added for resetting secret
- language: language added to inform user about resetting secret
- frontend: UI updated to allow user to reset secret

Side Effects
=========
None

Testing
----------
1. Make dev app through UI
- Configure https://github.com/abought/osf_oauth2_demo with credentials, ensure basic functionality
    - Note: above demo project currently not functional due to recent changes to cas / scopes
- Reset secret with UI, don't update credentials yet
- Ensure lack of functionality
- Update secret, reauthorize
- Ensure basic functionality

[#OSF-4351]
